src/openrct2/localisation/Formatter.h

```diff
@@ -77,16 +77,16 @@ class Formatter
 
         // clang-format off
         static_assert(
-            std::is_same_v<TSpecified, char*> ||
-            std::is_same_v<TSpecified, const char*> ||
-            std::is_same_v<TSpecified, int16_t> ||
-            std::is_same_v<TSpecified, int32_t> ||
-            std::is_same_v<TSpecified, money32> ||
-            std::is_same_v<TSpecified, rct_string_id> ||
-            std::is_same_v<TSpecified, uint16_t> ||
-            std::is_same_v<TSpecified, uint32_t> ||
-            std::is_same_v<TSpecified, utf8*> ||
-            std::is_same_v<TSpecified, const utf8*>
+            std::is_same_v<typename std::remove_cv<TSpecified>::type, char*> ||
+            std::is_same_v<typename std::remove_cv<TSpecified>::type, const char*> ||
+            std::is_same_v<typename std::remove_cv<TSpecified>::type, int16_t> ||
+            std::is_same_v<typename std::remove_cv<TSpecified>::type, int32_t> ||
+            std::is_same_v<typename std::remove_cv<TSpecified>::type, money32> ||
+            std::is_same_v<typename std::remove_cv<TSpecified>::type, rct_string_id> ||
+            std::is_same_v<typename std::remove_cv<TSpecified>::type, uint16_t> ||
+            std::is_same_v<typename std::remove_cv<TSpecified>::type, uint32_t> ||
+            std::is_same_v<typename std::remove_cv<TSpecified>::type, utf8*> ||
```

Comment:
Won't this fall on line 96/100 if the type is const?

Reply:
Nope, removing `const` and `volatile` does not change the size nor does it affect `std::is_integral`.

Somewhat related note: It is actually possible to pass a static const to the function while specifying a non-const type, but since the function doesn't write to the argument, it doesn't matter.

Reply:
Do we still require all of the is_same_v's then for the const char* and const utf8* now that we remove cv i think they are not required.

Reply:
Yes they are needed, since these are pointers, not types. std::remove_const/cv would make it a mutable pointer type.

```cpp
std::is_same_v<const char*, char*>; // False
std::is_same_v<char* const, char*>; // False
std::is_same_v<typename std::remove_cv<const char*>::type, char*>; // False
std::is_same_v<typename std::remove_cv<char* const>::type, char*>; // True
```